node/src/components/consensus/highway_core/state.rs

```diff
@@ -788,9 +788,20 @@ impl<C: Context> State<C> {
 
     /// Returns `true` if the `bhash` is a block that can have no children.
     pub(crate) fn is_terminal_block(&self, bhash: &C::Hash) -> bool {
+        let unit = self.unit(bhash);
+        let is_late_enough_after_first_block = self
+            .find_ancestor(bhash, 0)
+            .map(|first_block| {
+                unit.timestamp
+                    >= self
+                        .params
+                        .one_max_round_after(self.unit(first_block).timestamp)
```

Comment:
I'd find `max_round_len` a more natural `Params` method than `one_max_round_after`:
```suggestion
                    >= self.unit(first_block).timestamp + self.params.max_round_len()
```